openapi.yaml

```diff
@@ -0,0 +1,675 @@
+# This is the Open API interface for Open Service Broker API. Every attempt will
+# be made to make the Open API version of OSB API accurately represent the
+# written specification. If the spec and this document conflict, the spec is
+# the authroity.
+
+openapi: "3.0.0"
+
+info:
+  title: Open Service Broker API
+  description: The Service Broker API defines an HTTP(S) interface between Platforms and Service Brokers.
+  license: 
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  contact: 
+    name: Open Service Broker API
+    url: https://www.openservicebrokerapi.org/
+    email: open-service-broker-api@googlegroups.com
+  version: v2.13
+
+servers:
+  - url: http://example.com
+    description: Broker Endpoint
+  - url: http://localhost:3000
+    description: OSB Checker - MockOSB
+
+paths:
+  /v2/catalog:
+    get:
+      summary: get the service catalog
+      tags:
+        - Catalog
+      operationId: catalog.get
+      parameters:
+        - $ref: '#/components/parameters/APIVersion'
+      responses:
+        '200':
+          description: catalog response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Catalog'
+
+  /v2/service_instances/{instance_id}:
+    put:
+      summary: provision a service instance
+      tags:
+        - ServiceInstances
+      operationId: serviceInstance.provision
+      parameters:
+        - $ref: '#/components/parameters/APIVersion'
+        - $ref: '#/components/parameters/OriginatingIdentity'
+        - name: instance_id
+          in: path
+          description: instance id of instance to provision
+          required: true
+          schema:
+            type: string
+        - name: accepts_incomplete
+          in: query
+          description: asynchronous operations supported
+          schema:
+            type: boolean
+      requestBody:
+        description: parameters for the requested service instance provision
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServiceInstanceProvisionRequest'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ServiceInstanceProvision'
+        '201':
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ServiceInstanceProvision'
+        '202':
+          description: Accepted
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AsyncOperation'
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Object'
```

Comment:
I'm not 100% sure, but I think `error` and `description` field should be allowed to be returned with a 400

Reply:
This is what the spec says: 

> 400 -> MUST be returned if the request is malformed or missing mandatory data.

And then

> Service Brokers can include a user-facing message in the description field;

I struggled with this TBH, the spec says some responses should `{}` some should be in the `{ "error":"" , "description":""}` form and some it does not say. The line below [the table](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#response-2) says the quote above but it is not clear is that applies to all or some.

So I opted to define the response as Object at least, and where the spec calls it out, define the response as Error. 

Perhaps this is something we update after the doc has some usage if it is not working for folks or is unclear.

Reply:
Sounds good. If we can use this PR as an opportunity to clear up ambiguities in the spec, then I'd love for any error to be able to return the defined error object :)